### ISSUE TITLE
Autoalign plotting of maps is off by half a pixel

### ISSUE THREAD

**Issue by ayshih** [sunpy/sunpy#6794]
If you don't supply the pixel edges explicitly to `pcolormesh()`, it treats integer pixels as pixel edges, but we need integer pixels as pixel centers.  We just need to supply pixel edges in this call:
https://github.com/sunpy/sunpy/blob/b18bd5ec29953a877f2ae641ec30add87f680a79/sunpy/map/mapbase.py#L2549

**Comment by ayshih** [comment]
The correct call should be:
```python
ret = axes.pcolormesh(np.arange(data.shape[1] + 1) - 0.5,
                      np.arange(data.shape[0] + 1) - 0.5,
                      data, **imshow_args)
```
Also, a figure test needs to be updated/enhanced.

**Comment by Mihan786Chistie** [comment]
[FIX]Autoalign plotting of maps is off by half a pixel[https://github.com/sunpy/sunpy/pull/6796](url)

**Comment by Mihan786Chistie** [comment]
#6796 [FIX] Autoalign plotting of maps is off by half a pixel

**Comment by ayshih** [comment]
Fixed by #6796

Incidentally, I put this figure in that PR's discussion to show the bug:
![221392011-fb938cba-de69-44e6-a0ab-3b9237df3bd2](https://user-images.githubusercontent.com/991759/221933515-852864fa-7ce9-4db7-a95b-efc8b2bb5e6d.png)
